In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('country_vaccinations.csv')

In [3]:
df.head()

country iso_code        date  total_vaccinations  people_vaccinated  \
0  Albania      ALB  2021-01-10                 0.0                0.0   
1  Albania      ALB  2021-01-11                 NaN                NaN   
2  Albania      ALB  2021-01-12               128.0              128.0   
3  Albania      ALB  2021-01-13               188.0              188.0   
4  Albania      ALB  2021-01-14               266.0              266.0   

   people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
0                      NaN                     NaN                 NaN   
1                      NaN                     NaN                64.0   
2                      NaN                     NaN                64.0   
3                      NaN                    60.0                63.0   
4                      NaN                    78.0                66.0   

   total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                            0.00                           0.00   
1                             NaN                            NaN   
2                            0.00                           0.00   
3                            0.01                           0.01   
4                            0.01                           0.01   

   people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
0                                  NaN                             NaN   
1                                  NaN                            22.0   
2                                  NaN                            22.0   
3                                  NaN                            22.0   
4                                  NaN                            23.0   

          vaccines         source_name  \
0  Pfizer/BioNTech  Ministry of Health   
1  Pfizer/BioNTech  Ministry of Health   
2  Pfizer/BioNTech  Ministry of Health   
3  Pfizer/BioNTech  Ministry of Health   
4  Pfizer/BioNTech  Ministry of Health   

                                      source_website  
0  https://shendetesia.gov.al/covid19-ministria-e...  
1  https://shendetesia.gov.al/covid19-ministria-e...  
2  https://shendetesia.gov.al/covid19-ministria-e...  
3  https://shendetesia.gov.al/covid19-ministria-e...  
4  https://shendetesia.gov.al/covid19-ministria-e...

In [4]:
df.isnull().sum()

country                                   0
iso_code                                304
date                                      0
total_vaccinations                     1519
people_vaccinated                      1952
people_fully_vaccinated                2773
daily_vaccinations_raw                 1968
daily_vaccinations                      154
total_vaccinations_per_hundred         1519
people_vaccinated_per_hundred          1952
people_fully_vaccinated_per_hundred    2773
daily_vaccinations_per_million          154
vaccines                                  0
source_name                               0
source_website                            0
dtype: int64

### Dropping the rows with null values

In [5]:
df = df.dropna()

In [6]:
df.isnull().sum()

country                                0
iso_code                               0
date                                   0
total_vaccinations                     0
people_vaccinated                      0
people_fully_vaccinated                0
daily_vaccinations_raw                 0
daily_vaccinations                     0
total_vaccinations_per_hundred         0
people_vaccinated_per_hundred          0
people_fully_vaccinated_per_hundred    0
daily_vaccinations_per_million         0
vaccines                               0
source_name                            0
source_website                         0
dtype: int64

In [7]:
df = df.reset_index()

In [8]:
df.head()

index    country iso_code        date  total_vaccinations  \
0     39    Albania      ALB  2021-02-18              3049.0   
1    134  Argentina      ARG  2021-01-21            265724.0   
2    135  Argentina      ARG  2021-01-22            279602.0   
3    136  Argentina      ARG  2021-01-23            288064.0   
4    137  Argentina      ARG  2021-01-24            292023.0   

   people_vaccinated  people_fully_vaccinated  daily_vaccinations_raw  \
0             2438.0                    611.0                  1348.0   
1           249372.0                  16352.0                 17791.0   
2           254456.0                  25146.0                 13878.0   
3           258876.0                  29188.0                  8462.0   
4           260036.0                  31987.0                  3959.0   

   daily_vaccinations  total_vaccinations_per_hundred  \
0               254.0                            0.11   
1             11704.0                            0.59   
2             11263.0                            0.62   
3             11124.0                            0.64   
4             10342.0                            0.65   

   people_vaccinated_per_hundred  people_fully_vaccinated_per_hundred  \
0                           0.08                                 0.02   
1                           0.55                                 0.04   
2                           0.56                                 0.06   
3                           0.57                                 0.06   
4                           0.58                                 0.07   

   daily_vaccinations_per_million         vaccines         source_name  \
0                            88.0  Pfizer/BioNTech  Ministry of Health   
1                           259.0        Sputnik V  Ministry of Health   
2                           249.0        Sputnik V  Ministry of Health   
3                           246.0        Sputnik V  Ministry of Health   
4                           229.0        Sputnik V  Ministry of Health   

                                      source_website  
0  https://shendetesia.gov.al/covid19-ministria-e...  
1  http://datos.salud.gob.ar/dataset/vacunas-cont...  
2  http://datos.salud.gob.ar/dataset/vacunas-cont...  
3  http://datos.salud.gob.ar/dataset/vacunas-cont...  
4  http://datos.salud.gob.ar/dataset/vacunas-cont...

In [9]:
df['Month'] = pd.to_datetime(df['date'], format="%Y/%m/%d").dt.month
df['Year'] = pd.to_datetime(df['date'], format="%Y/%m/%d").dt.year
df['Day'] = pd.to_datetime(df['date'], format="%Y/%m/%d").dt.day

In [10]:
df.drop(['date', 'source_name', 'source_website'], axis=1, inplace=True)

In [11]:
df.head()

index    country iso_code  total_vaccinations  people_vaccinated  \
0     39    Albania      ALB              3049.0             2438.0   
1    134  Argentina      ARG            265724.0           249372.0   
2    135  Argentina      ARG            279602.0           254456.0   
3    136  Argentina      ARG            288064.0           258876.0   
4    137  Argentina      ARG            292023.0           260036.0   

   people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
0                    611.0                  1348.0               254.0   
1                  16352.0                 17791.0             11704.0   
2                  25146.0                 13878.0             11263.0   
3                  29188.0                  8462.0             11124.0   
4                  31987.0                  3959.0             10342.0   

   total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                            0.11                           0.08   
1                            0.59                           0.55   
2                            0.62                           0.56   
3                            0.64                           0.57   
4                            0.65                           0.58   

   people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
0                                 0.02                            88.0   
1                                 0.04                           259.0   
2                                 0.06                           249.0   
3                                 0.06                           246.0   
4                                 0.07                           229.0   

          vaccines  Month  Year  Day  
0  Pfizer/BioNTech      2  2021   18  
1        Sputnik V      1  2021   21  
2        Sputnik V      1  2021   22  
3        Sputnik V      1  2021   23  
4        Sputnik V      1  2021   24

In [12]:
df['iso_code'].value_counts()

ISR    55
GBR    47
CHL    46
DEU    44
DNK    43
ROU    42
CZE    42
ITA    42
BGR    41
MEX    41
HUN    40
SVK    40
BEL    40
SVN    40
NOR    40
USA    39
AUT    39
PRT    37
LTU    36
MLT    34
POL    33
GRC    33
LVA    33
ARG    32
FRA    32
FIN    28
IMN    27
GIB    27
EST    27
IDN    24
ESP    24
OMN    23
BRA    22
LUX    19
IRL    17
TUR    16
IND    13
CHE    11
ZAF     9
SYC     8
MAR     7
ISL     6
SRB     4
SWE     4
CYM     3
HRV     2
ARE     2
FRO     1
ALB     1
Name: iso_code, dtype: int64

In [13]:
thisdict = {'ISR' : 1,
            'GBR' : 2,
            'CHL' : 3,
            'DEU' : 4,
            'DNK' : 5,
            'ROU' : 6,
            'ITA' : 7,
            'CZE' : 8,
            'MEX' : 9,
            'BGR' : 10,
            'HUN' : 11,
            'SVN' : 12,
            'SVK' : 13,
            'BEL' : 14,
            'NOR' : 15,
            'USA' : 16,
            'AUT' : 17,
            'PRT' : 18,
            'LTU' : 19,
            'MLT' : 20,
            'GRC' : 21,
            'LVA' : 22,
            'POL' : 23,
            'FRA' : 24,
            'ARG' : 25,
            'FIN' : 26,
            'IMN' : 27,
            'GIB' : 28,
            'EST' : 29,
            'IDN' : 30,
            'ESP' : 31,
            'OMN' : 32,
            'BRA' : 33,
            'LUX' : 34,
            'IRL' : 35,
            'TUR' : 36,
            'IND' : 37,
            'CHE' : 38,
            'ZAF' : 39,
            'SYC' : 40,
            'MAR' : 41,
            'ISL' : 42,
            'SWE' : 43,
            'SRB' : 44,
            'CYM' : 45,
            'ARE' : 46,
            'HRV' : 47,
            'FRO' : 48,
            'ALB' : 49}

In [14]:
df['iso_code']= df['iso_code'].replace(thisdict)

In [15]:
df.drop('country', axis=1, inplace=True)

In [16]:
df.head()

index  iso_code  total_vaccinations  people_vaccinated  \
0     39        49              3049.0             2438.0   
1    134        25            265724.0           249372.0   
2    135        25            279602.0           254456.0   
3    136        25            288064.0           258876.0   
4    137        25            292023.0           260036.0   

   people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
0                    611.0                  1348.0               254.0   
1                  16352.0                 17791.0             11704.0   
2                  25146.0                 13878.0             11263.0   
3                  29188.0                  8462.0             11124.0   
4                  31987.0                  3959.0             10342.0   

   total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                            0.11                           0.08   
1                            0.59                           0.55   
2                            0.62                           0.56   
3                            0.64                           0.57   
4                            0.65                           0.58   

   people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
0                                 0.02                            88.0   
1                                 0.04                           259.0   
2                                 0.06                           249.0   
3                                 0.06                           246.0   
4                                 0.07                           229.0   

          vaccines  Month  Year  Day  
0  Pfizer/BioNTech      2  2021   18  
1        Sputnik V      1  2021   21  
2        Sputnik V      1  2021   22  
3        Sputnik V      1  2021   23  
4        Sputnik V      1  2021   24

In [17]:
df.drop('vaccines', axis=1, inplace=True)

In [23]:
df.drop('index', axis=1, inplace=True)

In [24]:
df.head()

iso_code  total_vaccinations  people_vaccinated  people_fully_vaccinated  \
0        49              3049.0             2438.0                    611.0   
1        25            265724.0           249372.0                  16352.0   
2        25            279602.0           254456.0                  25146.0   
3        25            288064.0           258876.0                  29188.0   
4        25            292023.0           260036.0                  31987.0   

   daily_vaccinations_raw  daily_vaccinations  total_vaccinations_per_hundred  \
0                  1348.0               254.0                            0.11   
1                 17791.0             11704.0                            0.59   
2                 13878.0             11263.0                            0.62   
3                  8462.0             11124.0                            0.64   
4                  3959.0             10342.0                            0.65   

   people_vaccinated_per_hundred  people_fully_vaccinated_per_hundred  \
0                           0.08                                 0.02   
1                           0.55                                 0.04   
2                           0.56                                 0.06   
3                           0.57                                 0.06   
4                           0.58                                 0.07   

   daily_vaccinations_per_million  Month  Year  Day  
0                            88.0      2  2021   18  
1                           259.0      1  2021   21  
2                           249.0      1  2021   22  
3                           246.0      1  2021   23  
4                           229.0      1  2021   24

### We have completed data preprocessing. We will now do model training and testing.

In [42]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

In [43]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.30, random_state=101)

In [44]:
lm = LinearRegression()

In [45]:
lm.fit(X_train, y_train)

LinearRegression()

In [47]:
predictions = lm.predict(X_test)

### Now, we will see total vaccinations in India, by 15th March, 2021.

In [75]:
data = {'iso_code' : 37.00,
        'people_vaccinated' : 8044786.00,
        'people_fully_vaccinated' : 7668.00,
        'daily_vaccinations_raw' : 84807.00,
        'daily_vaccinations' : 325305.00,
        'total_vaccinations_per_hundred' : 0.58,
        'people_vaccinated_per_hundred' : 0.58,
        'people_fully_vaccinated_per_hundred' : 0.00,
        'daily_vaccinations_per_million' : 236.00,
        'Month' : 3.00,
        'Year' : 2021.00,
        'Day' : 13.00}

In [76]:
india_data = pd.Series(data=data)

In [77]:
india_data

iso_code                                    37.00
people_vaccinated                      8044786.00
people_fully_vaccinated                   7668.00
daily_vaccinations_raw                   84807.00
daily_vaccinations                      325305.00
total_vaccinations_per_hundred               0.58
people_vaccinated_per_hundred                0.58
people_fully_vaccinated_per_hundred          0.00
daily_vaccinations_per_million             236.00
Month                                        3.00
Year                                      2021.00
Day                                         13.00
dtype: float64

In [78]:
totalVaccinationInIndia = lm.predict([india_data])

In [79]:
totalVaccinationInIndia

array([8059726.47285264])

## Here, we can see that total vaccinations in India by 15th March 2021 is around 8059726.47285264.